addons/website_event/tests/test_event_internals.py

```diff
@@ -190,6 +190,51 @@ def test_registration_answer_search(self):
         # should fetch "registration_3" because the answer to the third question is "Answer2" (as free text)
         self.assertEqual(search_res, registration_2 | registration_3)
 
+    def test_question_and_answer_access(self):
+        """ Check that some user groups have access to questions and answers only if they are linked to at
+        least one published event. """
+        question = self.env['event.question'].create({
+            "title": "Question",
+            "event_ids": [Command.create({
+                'name': 'Unpublished Event',
+                'website_published': False,
+            })]
+        })
+        answer = self.env['event.question.answer'].create({
+            "name": "Answer",
+            "question_id": question.id,
+        })
+        restricted_users = [self.user_employee, self.user_portal, self.user_public]
+        unrestricted_users = [self.admin_user, self.user_eventmanager, self.user_eventuser]
+
+        # Check that some user groups have no access to questions and answers of unpublished events.
+        for user in restricted_users:
+            self.assertFalse(question.with_user(user).has_access('read'),
+                             f'{user.name} should not have access to questions of unpublished events')
+            self.assertFalse(answer.with_user(user).has_access('read'),
+                             f'{user.name} should not have access to answers of unpublished events')
+
+        # Check that some user groups have access to questions and answers of unpublished events.
+        for user in unrestricted_users:
+            self.assertTrue(question.with_user(user).has_access('read'),
+                             f'{user.name} should have access to questions of unpublished events')
+            self.assertTrue(answer.with_user(user).has_access('read'),
+                             f'{user.name} should have access to answers of unpublished events')
```

Comment:
`of unpublished events` -> `only linked to unpublished events`  and since it is quite explicit, let us remove the comments above the loops :) :feather: